adyenv6core/src/com/adyen/v6/facades/DefaultAdyenCheckoutFacade.java

```diff
@@ -407,28 +407,36 @@ public PaymentsResponse handleRedirectPayload(HashMap<String, String> details) t
 
         String orderCode = response.getMerchantReference();
         OrderModel orderModel = retrievePendingOrder(orderCode);
+        updateOrderPaymentStatusAndInfo(orderModel, response);
 
-        if (PaymentsResponse.ResultCodeEnum.RECEIVED == response.getResultCode() || PaymentsResponse.ResultCodeEnum.AUTHORISED == response.getResultCode()) {
-            updateOrderPaymentStatusAndInfo(orderModel, OrderStatus.PAYMENT_AUTHORIZED, response);
-        } else {
-            updateOrderPaymentStatusAndInfo(orderModel, OrderStatus.CANCELLED, response);
+        if (PaymentsResponse.ResultCodeEnum.AUTHORISED != response.getResultCode()
+                && PaymentsResponse.ResultCodeEnum.RECEIVED != response.getResultCode()) {
             restoreCartFromOrder(orderCode);
         }
 
         return response;
     }
 
-    @Override
-    public void updateOrderPaymentStatusAndInfo(OrderModel orderModel, OrderStatus newStatus, PaymentsResponse paymentsResponse) {
-        //update payment info
-        getAdyenTransactionService().createPaymentTransactionFromResultCode(orderModel,
-                orderModel.getCode(),
-                paymentsResponse.getPspReference(),
-                paymentsResponse.getResultCode());
-
-        //update status
-        orderModel.setStatus(newStatus);
-        orderModel.setStatusInfo(paymentsResponse.getPspReference() + " - " + paymentsResponse.getResultCode().getValue());
+    private void updateOrderPaymentStatusAndInfo(OrderModel orderModel, PaymentsResponse paymentsResponse) {
+        PaymentsResponse.ResultCodeEnum resultCode = paymentsResponse.getResultCode();
+
+        if(PaymentsResponse.ResultCodeEnum.RECEIVED != resultCode) {
+            //payment authorisation is finished, update payment info
+            getAdyenTransactionService().createPaymentTransactionFromResultCode(orderModel,
+                    orderModel.getCode(),
+                    paymentsResponse.getPspReference(),
+                    paymentsResponse.getResultCode());
+        }
+
+        if(PaymentsResponse.ResultCodeEnum.AUTHORISED == resultCode || PaymentsResponse.ResultCodeEnum.RECEIVED == resultCode) {
+            //remove PAYMENT_PENDING status, will be processed by order management
+            orderModel.setStatus(null);
```

Comment:
if ResultCode is AUTHORISED, we should mark status. Not sure why it is null.

Reply:
As we talked, `PAYMENT_AUTHORIZED` will be set later by order management/fulfilment process. 
If we set this status now, it will change a few times before returning to that, which is confusing. It also doesn't make sense to change to it when the result code is RECEIVED.